### DISCUSSION TITLE
Do you plan to bring dashboard UI to docker?

### DISCUSSION THREAD

**Discussion by heyarviind** [supabase/supabase#1001] [Questions] [answered]
I tried to self-host supabase and find out that it did not have the UI and then I got confused about how to add tables, use APIs, etc.

Does superbase have any plans to add a dashboard to the self-host docker container?

**Comment by arladmin** [comment]
I have the same question as well.
Installed Supabase locally, today.

**Comment by kiwicopple** [comment, marked as answer]
> Does superbase have any plans to add a dashboard to the self-host docker container?

yes - you’ll see we *just* launched a [UI library](https://supabase.io). This is the first step for the local Dashboard

some background - when we started Supabase we didn’t know if people would like it, so we were shipping as fast as we could (it was just 2 of us). Now the usage, billing, user auth, and apis are all mixed in with the Dashboard. we’re pulling out the UI into separate components (and converting to typescript). you probably noticed the Dashboard improving as @MildTomato has been [adding components](https://github.com/supabase/ui/commits/develop) to this library. We’ll piece together the Local Dashboard again using this. 

we'd love help with this if anyone wants to jump into the UI library. also, please be patient with us on this one - we have been [shipping like mad](https://supabase.io/blog/2021/03/25/launch-week), and we still need to get Functions out (our next priority) and database migrations which are critical for local development

In the meantime, here are a bunch of awesome database tools that you can use (they also connect to your Supabase database):

- pgadmin: https://www.pgadmin.org (open source)
- dbeaver: https://dbeaver.com (this is what I use)
- BeeKeeper: https://beekeeperstudio.io - I haven't tried this but heard great things
- heidi: https://heidisql.com
- Table Plus: https://www.tableplus.io/

**Reply by heyarviind** [reply]
Thank you for the reply. We really understand what it feels like when you are not sure about the product and you just want to ship ASAP to get the feedback.

Supabase is an amazing product and we are continuously watching it 😉

Recently, I tried using [Hasura](https://hasura.io) for the same purpose and I felt a bit lost in setting up and managing the codebase. (Even though it provides the APIs on the go) and with Supabase it felt like everything is super easy to set up and we can get a fully working backend within minutes.

Also thank you for providing the links to various DB tools.

**Reply by ellis** [reply]
I know this is off-topic, but I'm also really excited about where Supabase is going.  Been using it in a side-project for a few weeks, and I'm now seriously considering moving my business's database from an older Google Cloud SQL postgres instance to supabase.  Speaking of Hasura, I was initially quite enthusiastic about it, but abandoned it after about a year after gradually coming to the conclusion that I'd much rather use JSON instead of graphql.

**Reply by kiwicopple** [reply]
Small update here - we have opened up the Grid: https://github.com/supabase/grid. We still have a lot more to do, but just wanted to ensure that we're checking in with the progress.

**Reply by AaronNGray** [reply]
Good to see your using TypeScript and your own React based UI library :)

**Reply by DavraYoung** [reply]
@kiwicopple 
Are Supabase team planning to make dashboard open source? If yes, could you please indicate when approximately? I found that [UI ](https://github.com/supabase/ui) components, Database [Grid](https://github.com/supabase/grid) is open.

**Reply by kiwicopple** [reply]
Yes we are :) 

You can follow the progress here: https://github.com/supabase/supabase/pull/2281

**Reply by itslevir** [reply]
I'd love to help out! What repo would I work with and how can I help?

**Reply by inian** [reply]
hi @lerichardson, you can contribute to the PR https://github.com/supabase/supabase/pull/2281 :)

**Reply by subwaymatch** [reply]
@kiwicopple Is there any update on the dashboard UI? I'm sure the Supabase team has their hands full, just want to ensure that dashboard UI is still on the roadmap. 👍 Thanks!

**Reply by inian** [reply]
stay tuned for our launch week end of this month ;)

**Reply by subwaymatch** [reply]
@inian Sounds super exciting!!!

**Reply by kiwicopple** [reply]
🎉 🚢 https://github.com/supabase/supabase/pull/4129
